### DISCUSSION TITLE
Highlighting bands in agile

### DISCUSSION THREAD

**Discussion by tre4b** [BottlecapDave/HomeAssistant-OctopusEnergy#1076] [Q&A]
I wanted to give my wife an idea of when Electricity was reasonable cost and when not.  my idea was to have an rgb light that set to a colour based on the current tariff.  I found your integration and have been trying to use it by defining multiple rolling selectors based around my bands.  It only seems to work for one band.  I wondered if there was an easier way to do this.  Maybe something built in that I have missed.

Ideally I want the light to go red when she does not have enough cheap time left for a 3hour was.  Currently it turns the light off when that is the case.  Which works.  She hates apps and things hence why trying to figure out something more basic.

**Comment by BottlecapDave** [comment]
Hello. It depends on what your consider cheap and theres so many different ways of doing it, which fully rely on your expected behaviour. It also depends on what you classify as enough cheap time.

The rolling target always evaluates x hour ahead, so the rates that will be reviewed will always be moving. The standard target rate sensor looks at a timeframe which moves ahead once the current timeframe is fully in the past.

The sensors go on when the cheap rates are current. There are also attributes on the sensor that determine when the current cheap rate period starts and ends. Based on what you've said, it sounds like you probably want to work off these start/end attributes using a template within your automation.

If you can give some more concrete examples with rates, what times in the rates you expect to be considered cheap (and for how long) and at what point you're wanting your light to come on.

**Comment by tre4b** [comment]
Hi,

I have set it up with a number of bands

Red = 0.3 and above
Purple = 0.25 to 0.3
Yellow = 0.2 to 0.25
Green = below 0.2
Blue=below zero.

Blue and Green override Yellow.   My challenge is that my wife is never sure when to put the devices on and I wanted to make it easy for her.

I have set the yellow light up to go on the following definition.

Hours required = 3 (enough time for a wash)
Hours as = Minimum
type=continuous
Looking ahead 8 hours.
Maximum rate 0.25
All existing target rates are in the past.

In my head the orange light should only be on if there are more than 3 hours below 0.25 during the next 8 hours such that there is time for a wash if ever the light is on.  Is that correct.

With the others I am setting the hours required to 30mins so it should always find the point where it applies but otherwise pretty much the same.  I feed these into toggles and then have an automation that looks at the toggles to determine what colour the light should be.  (you can have yellow, green and blue toggled on if the cost is less than zero for example).

Is there a way to view a future map of the costs?  It may be that getting some form of display with a dashboard might actually be more logical for this aspect.  For controlling things like the heaters and water heaters this is a great integration.

Trev

________________________________

**Reply by BottlecapDave** [reply]
The issue with solely using this is that the sensor will be on when it's half way through the target timeframe, which wouldn't be enough time for a wash. I personally use the target rate sensors to turn on devices at the correct time rather than require human intervention. I for instance use the remote start features on my washing machine and dishwasher. When these are activated, my Alexa will announce when these are due to be turned on (the first entry in the `target_times` attribute. If this is too late, we can then deactivate the remote start and start manually. Otherwise we let HA start the devices at the correct time. This obviously can only be done if you have devices that can be started remotely.

Another option would be to use the invert option on the target rate sensors to find the most expensive hours of the day and alert the house not to use these times. You could use the colours to alert when these times are approaching to avoid anything starting

```
{% set total_hours = (state_attr('binary_sensor.octopus_energy_rolling_target_xxx', 'target_times')[-1]["end"] - now()).total_seconds() / 60 / 60 %}
{%if total_hours >= 3 %}
Green
{%elif total_hours >= 1%}
Yellow
{%else%}
Red
{%endif%}
```

> Is there a way to view a future map of the costs

Future rates are spread across https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/entities/electricity/#current-day-rates and https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/entities/electricity/#next-day-rates. Because these are arrays, you can combine them and loop through them if you want to display them in a markdown card

```
| Start | End | Rate |
|-|-|-|
{% set rates = state_attr('event.octopus_energy_electricity_14p0125601_2200017705377_current_day_rates', 'rates') + state_attr('event.octopus_energy_electricity_14p0125601_2200017705377_next_day_rates', 'rates') %}{% if rates != None %}{% for rate in rates %}| {{ rate["start"] }} | {{ rate["end"] }} | {{ rate["value_inc_vat"] }} |
{% endfor %}{%endif%}
```

**Comment by tre4b** [comment]
Hi Dave,

Many thanks for this,  I like the idea of highlighting the most expensive hours and avoiding those.

Thank you so much for the feedback, I will be finding some time to play with those arrays as soon as I can.  Maybe I can connect them together with a custom indicator... Drat now I am going to waste a load of time on something again..  but I will enjoy it.

Trev

________________________________
